collector/src/main/java/com/bitmovin/analytics/BitmovinAnalytics.java

```diff
@@ -86,6 +92,31 @@ public void detachPlayer() {
         eventDataDispatcher.disable();
     }
 
+    private void addDeviceOrientationListener() {
+
+        SensorEventListener m_sensorEventListener = new SensorEventListener() {
+            private boolean isPortrait = Util.getDeviceOrientation(context).equals("portrait");
+
+            @Override
+            public void onSensorChanged(SensorEvent event) {
+                if ((playerStateMachine.getCurrentState() == PlayerState.PLAYING || playerStateMachine.getCurrentState() == PlayerState.PAUSE)
+                        && isPortrait != Util.getDeviceOrientation(context).equals("portrait")) {
+                    PlayerState originalState = playerStateMachine.getCurrentState();
+                    playerStateMachine.transitionState(PlayerState.ROTATE, getPosition());
+                    playerStateMachine.transitionState(originalState, getPosition());
+                    isPortrait = !isPortrait;
+                }
+            }
+            @Override
+            public void onAccuracyChanged(Sensor sensor, int accuracy) {
+                // No state change here
+            }
+        };
+
+        SensorManager sm = (SensorManager) context.getSystemService(SENSOR_SERVICE);
```

Comment:
Is this cast guaranteed to always succeed? What if `getSystemService` returns null?